Examples/Framework/include/ActsExamples/EventData/Measurement.hpp

```diff
@@ -166,6 +166,140 @@ class FixedSizeMeasurement {
   CovarianceMatrix m_cov;
 };
 
+template <typename indices_t>
+class VariableSizeMeasurement {
+  static constexpr std::size_t kFullSize =
+      Acts::detail::kParametersSize<indices_t>;
+
+  // TODO variable size
+  using Subspace = Acts::detail::VariableSizeSubspace<kFullSize>;
+
+ public:
+  using Scalar = Acts::ActsScalar;
+  /// Vector type containing for measured parameter values.
+  using ParametersVector = Eigen::Matrix<Scalar, Eigen::Dynamic, 1>;
+  using ParametersVectorMap = Eigen::Map<ParametersVector>;
+  using ConstParametersVectorMap = Eigen::Map<const ParametersVector>;
+  /// Matrix type for the measurement covariance.
+  using CovarianceMatrix =
```

Comment:
Instead of making these fully dynamic, this could use a dispatch mechanism to invoke functions with compile-time sizing.

Reply:
I adopted to what we do in `TrackStateProxy` calling the dynamic ones `effective` and added static size accessors.